src/classes/CW_AudienceCSVUploadController.cls

```diff
@@ -0,0 +1,313 @@
+/**
+* File Name   :    CW_AudienceCSVUploadController.cls
+* Description :    B-109507
+*                  upload CSv with groups/group memebers
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 02/04/2014 Klea Kolaric   Initial version
+* 0.2 29.04.2014 AleX Panka     Add shells Person Account support
+* 0.3 25.05.2014 Alex Panka     Audience groups rewards 
+* 0.4 08.08.2014 Cristina P		Upload Audience Group and Members in the same time
+*
+*/
+//Cristina P: 25/06/14: Changing the class from "with sharing" to "without sharing" 
+//so the user can create related records while cost/brief is in approval
+public without sharing class CW_AudienceCSVUploadController 
+{
+    
+    public final static List<Integer>  notAllowedChr= new List<Integer>{0,1,2,3,4,5,6,7,8,9,11,12,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31};
+	public transient Blob csvFile {get; set;}
+    public Id campaignId {get; set;}
+	public Id selectedSchId {get;set;}
+    public string filename{get;set;}
+    public static String scheduleLabel{get; set;}
+    public final Integer BATCH_SIZE = 2000;
+    public Id batchJobId;
+
+	public static Map<String,Id> contactsIdMap;
+	public static Map<String,Id> groupsIdMap;
+	public static Map<String,Id> agsIdMap;
+    
+    public List<Selectoption> getSheduleList(){
+    	  List<Selectoption>  result = new List<Selectoption> ();
+    	  result.Add( new Selectoption('','- none -'));
+    	  for (CW_Schedule__c rec : [SELECT ID ,Name FROM CW_Schedule__c WHERE Marketing_Campaign__c=:campaignId] ){
+    	  		result.Add( new Selectoption(rec.ID,rec.name));
+    	  }
+    	  return result;
+    }
+
+    public List<Selectoption> getRewardList(){
+    	  List<Selectoption>  result = new List<Selectoption> ();
+    	  result.Add( new Selectoption('','- none -'));
+    	  for (CW_Reward_Catalogue__c rec : [SELECT ID ,Name FROM CW_Reward_Catalogue__c WHERE Library__c<>null] ){
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.